docs/sources/flow/reference/components/prometheus.scrape.md

```diff
@@ -105,6 +107,36 @@ an `oauth2` block.
 
 {{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
 
+### clustering
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`enabled` | `bool` | Enables sharing targets with other cluster nodes.| `false` | yes
+
+When the agent is running in [clustered mode][], and `enabled` is set to true,
+then this `prometheus.scrape` component instance opts-in to participating in
+the cluster to distribute scrape load between all cluster nodes.
+
+Clustering assumes that all cluster nodes are running with the same
+configuration file, have access to the same service discovery APIs and that all
+`prometheus.scrape` components that have opted-in to using clustering are
+receiving the _exact same_ target set from upstream components in their
+`targets` argument.
+
+What happens is that all participating `prometheus.scrape` components use
+target labels and a consistent hashing algorithm to determine ownership for
+each of the targets between the cluster peers. Then, each peer only scrapes the
+subset of targets that it is responsible for, so that the scrape load is
+(nearly) evenly split. When a node joins or leaves the cluster, every peer
+re-calculates ownership and continues scraping with the new target set. This
+performs better than hashmod sharding where _all_ nodes have to be
```

Comment:
Worth noting drawbacks compared to hashmod? E.g. I suppose eventual consistency vs full consistency?

Reply:
Yes, that's the main argument. Adding it as a note here.